src/System.Runtime/tests/System/Double.cs

```diff
@@ -1,330 +1,402 @@
+                }
+                if (expected > 0)
+                {
+                    Assert.True(d1 > d2);
+                    Assert.False(d1 <= d2);
+                }
+                if (expected <= 0)
+                {
+                    Assert.True(d1 <= d2);
+                    Assert.False(d1 > d2);
+                }
+                if (expected < 0)
+                {
+                    Assert.True(d1 < d2);
+                    Assert.False(d1 >= d2);
+                }
+            }
+        }
+        IComparable comparable = d1;
+        Assert.Equal(expected, Math.Sign(comparable.CompareTo(value)));
     }
 
     [Fact]
-    public static void TestCompareToObjectInvalid()
+    public static void TestCompareTo_Invalid()
     {
         IComparable comparable = (double)234;
-        Assert.Throws<ArgumentException>(null, () => comparable.CompareTo("a")); //Obj is not a double
-    }
-
-    [Theory]
-    [InlineData((double)789, true)]
-    [InlineData((double)(-789), false)]
-    [InlineData((double)0, false)]
-    public static void TestEqualsObject(object obj, bool expected)
-    {
-        double i = 789;
-        Assert.Equal(expected, i.Equals(obj));
+        Assert.Throws<ArgumentException>(null, () => comparable.CompareTo((float)234)); // Obj is not a double
+        Assert.Throws<ArgumentException>(null, () => comparable.CompareTo("234")); // Obj is not a double
     }
 
     [Theory]
     [InlineData((double)789, (double)789, true)]
-    [InlineData((double)789, (double)(-789), false)]
+    [InlineData((double)789, (double)-789, false)]
     [InlineData((double)789, (double)0, false)]
     [InlineData(double.NaN, double.NaN, true)]
-    public static void TestEquals(double i1, double i2, bool expected)
-    {
-        Assert.Equal(expected, i1.Equals(i2));
-    }
-
-
-    [Fact]
-    public static void TestGetHashCode()
-    {
-        double i1 = 123;
-        double i2 = 654;
-
-        Assert.NotEqual(0, i1.GetHashCode());
-        Assert.NotEqual(i1.GetHashCode(), i2.GetHashCode());
-    }
-
-    [Fact]
-    public static void TestToString()
+    [InlineData((double)789, (float)789, false)]
+    [InlineData((double)789, "789", false)]
+    public static void TestEquals(double d1, object value, bool expected)
     {
-        double i1 = 6310;
-        Assert.Equal("6310", i1.ToString());
-
-        double i2 = -8249;
-        Assert.Equal("-8249", i2.ToString());
+        if (value is double)
+        {
+            double d2 = (double)value;
+            Assert.Equal(expected, d1.Equals(d2));
+
+            if (double.IsNaN(d1) && double.IsNaN(d2))
+            {
+                Assert.Equal(!expected, d1 == d2);
+                Assert.Equal(expected, d1 != d2);
+            }
+            else
+            {
+                Assert.Equal(expected, d1 == d2);
+                Assert.Equal(!expected, d1 != d2);
+            }
+
+            Assert.NotEqual(0, d1.GetHashCode());
```

Comment:
Is this supposed to be d2.GetHashCode()?  Or if not, why is this only done when value is double?